firstvoices/backend/views/custom_order_recalculate_views.py

```diff
@@ -0,0 +1,81 @@
+import rules
+from celery.result import AsyncResult
+from django.core.exceptions import PermissionDenied
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from backend.models import CustomOrderRecalculationPreviewResult, Site
+from backend.tasks.alphabet_tasks import recalculate_custom_order_preview
+
+
+class CustomOrderRecalculatePreviewView(APIView):
+    task_id = None
+
+    @staticmethod
+    def check_superadmin_status(request):
+        if not rules.has_perm("views.has_custom_order_access", request.user):
+            raise PermissionDenied()
+
+    def get(self, request, site_slug: str):
+        # Check Superadmin status
+        self.check_superadmin_status(request)
+
+        result = (
+            CustomOrderRecalculationPreviewResult.objects.filter(site__slug=site_slug)
+            .order_by("-date")
+            .first()
+        )
+
+        # If there is no result and no task_id, then there are no recalculation results for this site
+        if not self.task_id and not result:
+            return Response(
```

Comment:
Would it make sense here to return the same "Not started" status as below instead of the 404? Since this is a status report, I think it could make sense to always return a result, and sometimes the result is just "nothing to see here"

Reply:
I think if you skipped this conditional, the code below would still work and make sense for this case